### ISSUE TITLE
The road to 8.15

### ISSUE THREAD

**Issue by kleisauke** [libvips/libvips#3718]
`8.15` was just branched:
https://github.com/libvips/libvips/tree/8.15

Website updated in this branch:
https://github.com/libvips/libvips/tree/gh-pages-8.15

Release notes here:
https://www.libvips.org/2023/10/10/What's-new-in-8.15.html

TODO:
- [x] Propose v8.15.0-rc1 - https://github.com/libvips/libvips/pull/3719
  - [x] Update [`libvips/build-win64-mxe`](https://github.com/libvips/build-win64-mxe) for `-rc1` - https://github.com/libvips/build-win64-mxe/releases/tag/v8.15.0-rc1
  - [x] Open PR at [`lovell/sharp-libvips`](https://github.com/lovell/sharp-libvips) for `-rc1` - https://github.com/lovell/sharp-libvips/pull/198
- [x] Test and update downstream bindings.
  https://github.com/kleisauke/net-vips/compare/master...vips-8.15
  https://github.com/kleisauke/wasm-vips/compare/master...vips-8.15

**Comment by jcupitt** [comment]
I tried to improve the release notes a little. I updated pyvips and ruby-vips.

**Comment by remicollet** [comment]
Can we please revert da5cb2206a ?

**Comment by jcupitt** [comment]
Eeek! I'll look at the `dzsave` issue.

**Comment by kleisauke** [comment]
I'm not sure if this is the correct approach, but this patch for `dzsave` seems to fix it for me:
```diff
--- a/libvips/foreign/dzsave.c
+++ b/libvips/foreign/dzsave.c
@@ -2087,8 +2087,10 @@ vips_foreign_save_dz_build(VipsObject *object)
 
 		// and scale up by 2^depth ... we'll have complete tiles in every
 		// level
-		width *= 1 << level->n;
-		height *= 1 << level->n;
+		if (dz->depth == VIPS_FOREIGN_DZ_DEPTH_ONETILE) {
+			width *= 1 << level->n;
+			height *= 1 << level->n;
+		}
 
 		level_free(level);
```

**Comment by kleisauke** [comment]
... work-in-progress commit https://github.com/kleisauke/libvips/commit/66e9b9635b0cb786adfcbf789dfffd4e68f57238 fixes a couple of other `dzsave` bugs found in sharp' test suite. The only remaining one is:
```
  1) Tile
       Google layout with depth onepixel:

      Uncaught AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:

13 !== 9

      + expected - actual

      -13
      +9
      
      at assertGoogleTiles (test/unit/tile.js:76:10)
      at /home/kleisauke/sharp/test/unit/tile.js:761:13
      at Object.<anonymous> (lib/output.js:576:258)
```
https://github.com/lovell/sharp/blob/069d1786f5061db063dcdea41bb6aa42b000db06/test/unit/tile.js#L761

Which might indicate that the above patch is incorrect.

**Comment by jcupitt** [comment]
I pushed Sergey's heifload fix to 8.15 ... sorry, I should have made a PR, I wasn't thinking.

**Comment by kleisauke** [comment]
No worries, I'm glad to see it fixed. However, I'm not sure if we should do another release in the 8.14.x series.

**Comment by jcupitt** [comment]
I agree, we don't have enough need for another 8.14 release, at least for now.

**Comment by jcupitt** [comment]
I had a look at the google layout + one pixel combination again and it won't ever work with the current code. I'll have another go.

**Comment by jcupitt** [comment]
Thinking about it (again) maybe onepixel and google mode doesn't make sense as a combination?

In onepixel mode, the smallest pyramid level is 1x1 pixel, but gmaps mode has no incomplete tiles (edge tiles are padded to tile boundaries). So you can't have both -- the smallest level is always going to be bigger than 1x1 pixel.

A very simple fix would be to check for gmaps + onepixel and change it to onetile instead. Any thoughts?

**Comment by kleisauke** [comment]
> A very simple fix would be to check for gmaps + onepixel and change it to onetile instead. Any thoughts?

This fix sounds good to me.

**Comment by jcupitt** [comment]
Sorry for the delay. I pushed https://github.com/libvips/libvips/commit/12c2100154d51e0a025ee320a9939811753b568b to the 8.15 branch, so in gmaps mode we now force onepixel mode to onetile.

**Comment by jcupitt** [comment]
... and polished the release notes a little. Maybe we're ready for rc2?

**Comment by kleisauke** [comment]
Great! There are two open PRs both targeting the 8.15 branch (i.e. #3731 and #3735), so we may have to wait a bit before tagging rc2.

**Comment by kleisauke** [comment]
I think we're ready for rc2 now, I just opened PR #3745.

**Comment by lovell** [comment]
The performance improvement in switching from orc to highway exposed a couple of race conditions in the sharp unit tests :sweat_smile: but all passing now with v8.15.0-rc2. https://github.com/lovell/sharp/actions/runs/6754585967

**Comment by kleisauke** [comment]
NetVips tests are passing on all platforms/archs using v8.15.0-rc2. :heavy_check_mark:
https://www.nuget.org/packages/NetVips.Native/8.15.0-rc2
https://github.com/kleisauke/net-vips/actions/runs/6761129756
https://ci.appveyor.com/project/kleisauke/net-vips/builds/48445347

wasm-vips test suite also passes using v8.15.0-rc2 (both on the [`vips-8.15`](https://github.com/kleisauke/wasm-vips/tree/vips-8.15) and [`simd-highway`](https://github.com/kleisauke/wasm-vips/tree/simd-highway) branch). :heavy_check_mark:
https://github.com/kleisauke/wasm-vips/actions/runs/6755186066
https://github.com/kleisauke/wasm-vips/actions/runs/6755186276

No issues were found during the A/B test on the [`weserv/images`](https://github.com/weserv/images) project using v8.15.0-rc1, so v8.15.0-rc2 is now deployed in production. :heavy_check_mark:

**Comment by jcupitt** [comment]
\o/ all my downstream projects look good too!

**Comment by kleisauke** [comment]
I think we're ready for v8.15.0 final,  I just opened PR #3752 and published the release notes:
https://www.libvips.org/2023/10/10/What's-new-in-8.15.html

**Comment by kleisauke** [comment]
All done. Thank you everyone!

**Comment by ur5us** [comment]
@lovell 

> AVIF encoding fails with heif_writer callback returned a null error text (5.2001), which relates to the https://github.com/strukturag/libheif/commit/c95ed71a67003a49ddfe2e1bf810bfc96e61d333 change. It looks like heifsave will need to detect and use the new and confusingly-named heif_error_success API - I can work on a patch for this.

Curious, has that been addressed in the 8.15 release?

**Comment by lovell** [comment]
@ur5us Yes, via commit https://github.com/libvips/libvips/commit/dda2df2081b48300954900d23082d19195e4356d
